lib/mix/lib/mix/dep/loader.ex

```diff
@@ -320,6 +322,13 @@ defmodule Mix.Dep.Loader do
     end
   end
 
+  defp incompatible_requirement?(%Mix.Dep{opts: opts}) do
+    case opts[:lock] do
+      {scm, _, _, _} -> is_nil(opts[scm])
```

Comment:
Ok, thanks for the explanation. I will close this PR for now then.
I am going to see if I find a better way to handle this.